lib/engine-object-file/README.md

```diff
@@ -0,0 +1,186 @@
+# Wasmer Engine Object File
+
+This is an [engine](https://crates.io/crates/wasmer-engine) for the [wasmer](https://crates.io/crates/wasmer/1.0.0-alpha01.0) WebAssembly VM.
+
+This engine is used to produce native code that can be linked against providing
+a sandboxed WebAssembly runtime environment for the compiled module with no need
+for runtime compilation.
+
+## Example of use
+
+First we compile our WebAssembly file with Wasmer
+```sh
+wasmer compile path/to/wasm/file.wasm --llvm --object-file -o my_wasm.o --header my_wasm.h
+```
+
+You will then see output like:
+```
+Engine: objectfile
+Compiler: llvm
+Target: x86_64-apple-darwin
+✔ File compiled successfully to `my_wasm.o`.
+✔ Header file generated successfully at `my_wasm.h`.
+```
+
+Now lets create a program to link with this object file.
+
+```C
+#include "wasmer_wasm.h"
+#include "wasm.h"
+#include "my_wasm.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+void wasmer_function__1(void);
+void wasmer_trampoline_function_call__1(void*, void*, void*);
+
+// todo: add to wasmer_wasm.h or something
+void* wasm_instance_get_vmctx_ptr(wasm_instance_t*);
```

Comment:
In general, I'm a bit wary of exposing a function like this. I don't think we should expose the vmctx to the user, since it leaks internal structures

Reply:
I see what you're saying, but it's required to be able to call things directly and the vmctx is completely opaque from the perspective of the user. I'll get back to this after investigating the tradeoff of calling through the Wasm C API

Reply:
Let's add a comment in the code about this, ship that comment with the code as TODO and then add tests and merge the PR (once reviewed).